src/Windows/Avalonia.Win32/SimpleWindow.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using Avalonia.Win32.Interop;
+
+namespace Avalonia.Win32;
+
+internal class SimpleWindow : IDisposable
+{
+    private readonly UnmanagedMethods.WndProc? _wndProc;
+    private static UnmanagedMethods.WndProc s_wndProcDelegate;
+    public IntPtr Handle { get; private set; }
+    private static string s_className;
+    private static uint s_classAtom;
+    private static ConcurrentDictionary<IntPtr, SimpleWindow> s_Instances = new();
+
+    static SimpleWindow()
+    {
+        s_wndProcDelegate = WndProc;
+        var wndClassEx = new UnmanagedMethods.WNDCLASSEX
+        {
+            cbSize = Marshal.SizeOf<UnmanagedMethods.WNDCLASSEX>(),
+            hInstance = UnmanagedMethods.GetModuleHandle(null),
+            lpfnWndProc = s_wndProcDelegate,
+            lpszClassName = s_className = "AvaloniaSimpleWindow-" + Guid.NewGuid(),
```

Comment:
Not really descriptive. Having different classnames for offscreen and winui-hack windows might help somebody's in the future.

Reply:
The window class is now reused for all windows to avoid leaking resources for no reason. It's not possible to customize it per-window anymore